src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,134 @@
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useMemo, useState } from 'react';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { SortBy } from './enums/SortBy';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const preparedPeople = useMemo(() => {
+    let filteredPeople = people.map(person => {
+      const mother = people
+        .find(mom => person.motherName === mom.name) || null;
+
+      const father = people
+        .find(dad => person.fatherName === dad.name) || null;
+
+      return {
+        ...person,
+        mother,
+        father,
+      };
+    });
+
+    if (query) {
+      filteredPeople = filteredPeople.filter(person => (
+        person.name.toLowerCase().includes(query.toLowerCase())
+        || person.motherName?.toLowerCase().includes(query.toLowerCase())
+        || person.fatherName?.toLowerCase().includes(query.toLowerCase())
+      ));
+    }
+
+    if (centuries.length > 0) {
+      filteredPeople = filteredPeople.filter(person => centuries
+        .find(century => Math.ceil(person.born / 100) === +century));
+    }
+
+    if (sex) {
+      filteredPeople = filteredPeople.filter(person => person.sex === sex);
+    }
+
+    if (sort) {
+      filteredPeople = filteredPeople.sort((a, b) => {
+        switch (sort) {
+          case SortBy.name:
+            return order
+              ? b.name.localeCompare(a.name)
```

Comment:
```suggestion
              ? b[sort]localeCompare(a[sort])
```
it will allow you to join cases you handle in the same way: name and sex, born and died